icon_validator/rules/workflow_validators/workflow_use_case_validator.py

```diff
@@ -0,0 +1,77 @@
+import requests
+from icon_plugin_spec.plugin_spec import KomandPluginSpec
+
+from icon_validator.rules.validator import KomandPluginValidator
+from icon_validator.exceptions import ValidationException
+
+
+class WorkflowUseCaseValidator(KomandPluginValidator):
+    @staticmethod
+    def get_approved_usecase_tags(response_json: dict) -> [str]:
+        approved_usecases = []
+        for result in response_json["results"]:
+            if result["type"] == "usecase":
+                approved_usecases.append(result["name"])
+
+        return approved_usecases
+
+    @staticmethod
+    def get_approved_usecase_tags_with_paging() -> [str]:
```

Comment:
If you want a fun thing to try (not required for this PR, but could improve performance if this usecase list ever grew huge), you can turn this function into a [generator](https://wiki.python.org/moin/Generators) that yields usecases that can be compared against. You could then try comparing all the usecases in the workflow against the current set, and if they don't match, then get the next set from the generator (which would be the next page from the API), and continue until all possible usecase matches have been exhausted.